lib/devices/ios/ios.js

```diff
@@ -778,57 +784,33 @@ IOS.prototype.setSafariPrefs = function () {
   }
 };
 
-IOS.prototype.detectTraceTemplate = function (cb) {
-  logger.debug("Detecting automation tracetemplate");
-  var msg;
-  if (!this.args.automationTraceTemplatePath) {
-    xcode.getPath(function (err, xcodeFolderPath) {
-      if (err) return cb(err);
-      if (xcodeFolderPath !== null) {
-        var instExt = this.iOSSDKVersion >= 8 ? 'xrplugin' : 'bundle';
-        var xcodeTraceTemplatePath = path.resolve(xcodeFolderPath,
-          "../Applications/Instruments.app/Contents/PlugIns",
-          "AutomationInstrument." + instExt + "/Contents/Resources",
-          "Automation.tracetemplate");
-        if (fs.existsSync(xcodeTraceTemplatePath)) {
-          this.args.automationTraceTemplatePath = xcodeTraceTemplatePath;
+IOS.prototype.detectUdid = function (cb) {
+  if (this.args.udid !== null && this.args.udid === "auto") {
+    which('idevice_id', function (err, cmdPath) {
+      logger.debug("Auto-detecting iOS udid...");
+
+      var udidetectPath = cmdPath + " -l";
+      if (err) {
+        udidetectPath = require.resolve('udidetect');
+      }
+
+      var udiddetectProc = exec(udidetectPath, { maxBuffer: 524288, timeout: 3000 }, function (err, stdout) {
```

Comment:
I just ran a test, and it works as I expected. Do you mind cleaning this up while we're touching it?